crates/sui-adapter/src/adapter.rs

```diff
@@ -628,10 +628,9 @@ fn process_successful_execution<S: Storage + ParentSync>(
             None => match state_view.get_latest_parent_entry_ref(id) {
                 Ok(Some((_, previous_version, _))) => previous_version,
                 Ok(None) => {
-                    // TODO we don't really need to mark this as deleted, as the object was not
-                    // created this txn but has never existed in storage. We just need a better
-                    // way of detecting this rather than relying on the parent sync
-                    SequenceNumber::new()
+                    // This object was not created this transaction but has never existed in
+                    // storage, skip it.
+                    continue;
```

Comment:
isn't the Ok(Some(...)) case also questionable - if an object did exist in the store, and then was wrapped, and then deleted, is it approriate to use the last version that existed before being wrapped?

Reply:
also, i'm slightly concerned about how this could mask a bug in which parent_sync isn't updated, but maybe there's not much we can do about that.

Reply:
I think there's an argument to be made that because wrapping is considered a form of deletion, so if the object was not in `by_value_objects`, this deletion should be entirely ignored (i.e. advocating for getting rid of the use of parent sync altogether), but things are set-up this way because in practice, it's useful to know when the object is gone for good vs just hidden.  That is a deeper subject than the thesis of this PR though.

Reply:
> isn't the Ok(Some(...)) case also questionable - if an object did exist in the store, and then was wrapped, and then deleted, is it approriate to use the last version that existed before being wrapped?

I'm not sure what else you would suggest? 
IIRC correctly, we used to set it to `MAX - 1` instead of `previous_version`, but that feels even more confusing. 
If you conceptually think of the version as being a field of the object (even though that isn't the case in Move), the `previous_version` is the consistent behavior. 
Though definitely open to suggestions here.

> also, i'm slightly concerned about how this could mask a bug in which parent_sync isn't updated, but maybe there's not much we can do about that.

That is a potential issue, yes. One thing we have discussed doing is forming a write-ahead, crash-recovery log of `Txn => ObjectID => Version` which would indicate the version used as an input in this transaction in the case that this transaction is being reprocessed (for whatever reason). We could also populate that table for this parent sync case, which I think should help prevent any bugs with the parent_sync, but maybe

Reply:
> in the case that this transaction is being reprocessed

@tnowacki We won't have to worry about that - the current and future plan for execution recovery is that we only ever execute once, then right the temporary store + effects atomically to a recovery "log" before updating the store. If the store updates are interrupted, we re-read the recovery log rather than re-executing. Otherwise there is no way to be correct wrt dynamic child loading as far as I can see.